From 89f1a074e07fdd770622b8c57511c94bff1ef075 Mon Sep 17 00:00:00 2001
From: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Mon, 6 Sep 2021 11:57:46 -0300
Subject: [PATCH 037/500] [Backport] perf beauty: Update copy of linux/socket.h
 with the kernel sources

mainline inclusion
from mainline-v5.15-rc1
commit bb91de44693b1c10fe2f9e668506b01e88efed0e
category: bugfix
bugzilla: 188335

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?h=v6.2-rc7&id=bb91de44693b1c10fe2f9e668506b01e88efed0e

--------------------------------

To pick the changes in:

Fixes: d32f89da7fa8ccc8 ("net: add accept helper not installing fd")
Fixes: bc49d8169aa72295 ("mctp: Add MCTP base")

This automagically adds support for the AF_MCTP protocol domain:

  $ tools/perf/trace/beauty/socket.sh > before
  $ cp include/linux/socket.h tools/perf/trace/beauty/include/linux/socket.h
  $ tools/perf/trace/beauty/socket.sh > after
  $ diff -u before after
  $ --- before	2021-09-06 11:57:14.972747200 -0300
  $ +++ after	2021-09-06 11:57:30.541920222 -0300
  $ @@ -44,4 +44,5 @@
  $  	[42] = "QIPCRTR",
  $  	[43] = "SMC",
  $  	[44] = "XDP",
  $ +	[45] = "MCTP",
  $  };
  $

This will allow 'perf trace' to translate 45 into "MCTP" as is done with
the other domains:

  # perf trace -e socket*
     0.000 chronyd/1029 socket(family: INET, type: DGRAM|CLOEXEC|NONBLOCK, protocol: IP) = 4
  ^C#

This addresses this perf build warning:

  Warning: Kernel ABI header at 'tools/perf/trace/beauty/include/linux/socket.h' differs from latest version at 'include/linux/socket.h'
  diff -u tools/perf/trace/beauty/include/linux/socket.h include/linux/socket.h

Cc: David S. Miller <davem@davemloft.net>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Jeremy Kerr <jk@codeconstruct.com.au>
Cc: Pavel Begunkov <asml.silence@gmail.com>
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

Conflict:
	tools/perf/trace/beauty/include/linux/socket.h

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
---
 tools/perf/trace/beauty/include/linux/socket.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/perf/trace/beauty/include/linux/socket.h b/tools/perf/trace/beauty/include/linux/socket.h
index 42222a84167f..c3b35d18bcd3 100644
--- a/tools/perf/trace/beauty/include/linux/socket.h
+++ b/tools/perf/trace/beauty/include/linux/socket.h
@@ -421,6 +421,9 @@ extern int __sys_accept4_file(struct file *file, unsigned file_flags,
 			struct sockaddr __user *upeer_sockaddr,
 			 int __user *upeer_addrlen, int flags,
 			 unsigned long nofile);
+extern struct file *do_accept(struct file *file, unsigned file_flags,
+			      struct sockaddr __user *upeer_sockaddr,
+			      int __user *upeer_addrlen, int flags);
 extern int __sys_accept4(int fd, struct sockaddr __user *upeer_sockaddr,
 			 int __user *upeer_addrlen, int flags);
 extern int __sys_socket(int family, int type, int protocol);
-- 
2.17.1

